xterm-color.el

```diff
@@ -497,11 +532,18 @@ in LIFO order."
                                     xterm-color--current-bg
                                     (/= xterm-color--attributes 0)))
           (has? (attrib)       `(/= (logand ,attrib xterm-color--attributes) 0))
-          (face-cache-get ()   `(gethash (logior (ash xterm-color--attributes 16)
-                                                 (ash (or xterm-color--current-bg 0) 8)
-                                                 (or xterm-color--current-fg 0))
+          (face-cache-get ()   `(gethash (let ((fg (or xterm-color--current-fg 0))
+                                               (bg (or xterm-color--current-bg 0)))
+                                           (if (has? +24-bit+)
+                                               (logior (ash xterm-color--attributes 48)
+                                                       (ash bg 24)
+                                                       fg)
+                                             (logior (ash xterm-color--attributes 16)
+                                                     (ash bg 8)
+                                                     fg)))
                                          xterm-color--face-cache))
           (face! (k v)         `(setq plistf (plist-put plistf ,k ,v)))
+          (xterm-color-24-bit (color) `(format "#%06x" ,color))
```

Comment:
Rename to truecolor. It's inside a macrolet, no reason to add the xterm-color prefix.
Someone asked me to make xterm-color-256 part of the external API which is why it's not in the macrolet too.